api_gateway_srv/gateway/api/controller.py

```diff
@@ -0,0 +1,81 @@
+from .microservices_api import user_service, reminder_service
+
+def login_user(username, password):
+    user_srv = user_service()
+
+    # Get user 
+    response = user_srv.get_user(username)
+    if not response:
+        # If doesn't exist => register user
+        if not user_srv.add_user(username, password):
```

Comment:
I'm not sure I understand the design decision behind creating a user on the fly here whenever we try to login with a non-existing user.

Reply:
I know, I didn't like either. It was only to keep UI as simple as possible to solve other things first. If I have time I will add a Sign up UI and I could also add a Change Pwd UI. Anyway User microservice it has both features already implemented. So I should only add proper Api Gateway routes.